FF96 ElementInternals.willValidate - relnote and update

hamishwillee opened the pull request:
FF96 adds support for  [ElementInternals.willValidate](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals/willValidate) in https://bugzilla.mozilla.org/show_bug.cgi?id=1556365

This PR adds a release note and updates the docs for the property to current syntax. Also added a bit more information about things that might bar a custom element from returning true - seems relevant.

other docs work for this tracked in https://github.com/mdn/content/issues/10851#issuecomment-987594191

hamishwillee committed:
FF96 ElementInternals.willValidate - relnote and update

EdgarChen commented:
Note that the form-associated custom elements are put behind the pref `dom.webcomponents.formAssociatedCustomElement.enabled` which is enabled on Nightly only. Flipping on all channels by default is tracked by https://bugzilla.mozilla.org/show_bug.cgi?id=1745378.

hamishwillee commented:
Thanks very much @EdgarChen - I thought I'd checked that. Fixed in docs (removed from release note, added to experimental features) and in browser compat data - added preference info.
